crypto/crypto.go

```diff
@@ -53,20 +53,19 @@ func deriveCreate2Address(salt string, msgSender string, implementation string)
 
 	bytecode, _ := GetProxyCode(ensureNoPrefix(implementation))
 
-	input, _ := hex.DecodeString("ff" + ensureNoPrefix(msgSender) + salt + common.Bytes2Hex(crypto.Keccak256(bytecode)))
+	input, _ := hex.DecodeString("ff" + ensureNoPrefix(msgSender) + ensureNoPrefix(salt) + common.Bytes2Hex(crypto.Keccak256(bytecode)))
 	return "0x" + common.Bytes2Hex(crypto.Keccak256(input))[24:], nil
 }
 
 // GenerateChannelAddress generate channel address from given identity hash
-func GenerateChannelAddress(identity string, registry string, channelImplementation string) (address string, err error) {
+func GenerateChannelAddress(identity, accountant, registry string, channelImplementation string) (address string, err error) {
```

Comment:
All the fields are strings, so:

`func GenerateChannelAddress  (identity, accountant, registry, channelImplementation string)`